manual/manual/tutorials/coreexamples.etex

```diff
@@ -537,6 +548,39 @@ let temporarily_set_reference ref newval funct =
     raise x;;
 \end{caml_example}
 
+An alternative to "try"\ldots"with" is to catch the exception while
+pattern matching:
+\begin{caml_example}{toplevel}
+let assoc_opt x l =
+  match List.assoc x l with
+  | exception Not_found -> None
+  | x -> Some x;;
```

Comment:
This example would be more enlightening if the value branch involved some evaluation:

```ocaml
  | x -> f x
```

As it is, the example could be more concisely written using `try`:

```ocaml
let assoc_opt x l =
  try Some (List.assoc x l) with
   Not_found -> None
```

However, if the value branch involves evaluation then it's harder to write the function using `try`, since moving the call to `f` between `try` and `with` changes the exception-handling behaviour.

Reply:
Good idea, adding `f x` in the non-exceptional branch is a nice touch.